sources/github.go

```diff
@@ -433,49 +432,47 @@ func (g *githubSource) InitialTag(ctx context.Context, accessToken *AccessToken,
 	owner := repoPieces[0]
 	name := repoPieces[1]
 
+	client := g.graphqlFunc(ctx, accessToken.Token, accessToken.Type, g.cfg.RateLimitTimeoutSeconds, g.cfg.RateLimitRetryCount)
+
 	repo, err := githubClient.GetRepo(ctx, owner, name)
 	if err != nil {
 		return errors.Wrap(err, "failed to get repo")
 	}
 
-	tags, err := githubClient.ListRepoTags(ctx, owner, name, &github.ListOptions{})
-	if err != nil {
-		return errors.Wrapf(err, "failed to list tags for repo '%s/%s'", owner, name)
-	}
+	if commitSha == "" {
+		tags, err := githubClient.ListRepoTags(ctx, owner, name, &github.ListOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "failed to list tags for repo '%s/%s'", owner, name)
+		}
 
-	if len(tags) > 0 {
-		return nil
-	}
+		if len(tags) > 0 {
+			return nil
```

Comment:
This func basically only tags the repository with default tag if the repo doesn't have any tags at all. 

Shouldn't this be actually a fallback behavior ? Using this as a package I would expect to be able to specify the tag I want to create on the repository. 

From the comment I understand the intent was to have this as an initial tag creation mechanism. Would it make sense to change that now ?

Reply:
I wanted to make this function a Tag one but changed my mind along the way as it should stay InitialTag not Tag. I've changed the name back